cli/azd/pkg/convert/util.go

```diff
@@ -14,3 +14,12 @@ func ToValueWithDefault[T any](ptr *T, defaultValue T) T {
 func RefOf[T any](value T) *T {
 	return &value
 }
+
+// Attempts to convert the specified value to a string, otherwise returns the default value
+func ToStringWithDefault(value any, defaultValue string) string {
+	if str, ok := value.(string); ok {
+		return str
```

Comment:
If `any` is equal to empty string, do we want to return the `defaultValue` or the empty string?
It might be good to mention this case on the method's documentation/comments.

I also wonder is we could combine `ToValueWithDefault()` and this `ToStringWithDefault()` into one only method that handles pulling the value (when passing a pointer) or the directly taking the value (when not a pointer). That would make it easier for the caller.  For example, we can call `ToStringWithDefault(ptr, "")` and even if the `ptr` is referencing a string, we would get the `defaultValue`, just because the ptr can't be converted to string.

Reply:
Since an empty string is already a string then it would use the user defined value instead of the default value.

Reply:
We can't combine them 100% but for most use cases you'll be able to use `convert.ToValueWithDefault(...)`.  `ToStringWithDefault` is for use cases where we don't know the input type.  Its either defined as any and could be a string value or pointer to a string.  Keeping them both for now.